pypeit/spec2dobj.py

```diff
@@ -394,16 +421,16 @@ def build_primary_hdr(self, raw_header, spectrograph, master_key_dict=None, mast
         if redux_path is not None:
             hdr['PYPRDXP'] = redux_path
         # Sky sub mode
-        if self['meta']['ir_redux']:
+        if 'ir_redux' in self['meta'] and self['meta']['ir_redux']:
```

Comment:
I think that ir_redux and find_negative should always be in self['meta']. Did you encounter otherwise?

Reply:
They may always be in practice, but I don't know.  The main issue is that these items are not added when the object is instantiated, so it's possible this will throw an error.